### ISSUE TITLE
news app improvements

### ISSUE THREAD

**Issue by gyst** [ploneintranet/ploneintranet.prototype#92]
Full analysis of the current news app design uncovers a number of inconsistencies. I'm scheduling development work with specifications that assume the prototype will be updated with the following changes.

Prio is high since we will start developing this next Monday 8 June.

News app:
- Create news item does not launch editor, instead weird topnav change
- Create news item "Show on" radio please remove (section not set + inconsistent with edit view). Instead offer section select since this is a required field.
- Same weird topnav change for create section, which should NOT launch editor but instead should refresh sidepane to show new section
- Section/delete links to 404 confirmation-delete-news-item.htm (should be "section"!)
  MVP solution is to give a message (we have two messages: "You must first move all news items to another section before you can delete this section." and "This is the only news section and cannot be deleted. Create another section before deleting this one.")

Must read (are sorted on top in magazine/overview but only when unread):
- Consumer magazine/overview: indicator + "have read" setter 
- Editor magazine/overview indicator
- Editor edit checkbox

App navigation:
- Should sort on and show must-read status, publication date and publication state instead of description
  since the combination of these 3 parameters controls the display and sequence of news items
- Future publication date marked visually, whole item stands out as future published
- Unpublished state indicated, whole item stands out as unpublished, when undated marked (no date) sorted on top

Blockquote:
- Styled in "view" but not styled in "edit" mode - maybe not in prototype but ask JC to tweak Raptor

Breadcrumb is confusing if you click from "all news" into an item and then get "Back to press releases".
Better is to keep the section navbar with the current section highlighted.

**Comment by gyst** [comment]
NB "mustread" is low prio. Difficult to get that performant in the backend.

**Comment by cornae** [comment]
> News app:
> - Create news item does not launch editor, instead weird topnav change

Cannot reproduce:

![image](https://cloud.githubusercontent.com/assets/738601/8004281/c91a5cea-0b7e-11e5-97d8-41eebb60007f.png)

> - Create news item "Show on" radio please remove (section not set + inconsistent with edit view). Instead offer section select since this is a required field.
> - Same weird topnav change for create section, which should NOT launch editor but instead should refresh sidepane to show new section

Of course, but I didn't spend time on this level of simulation because of time and budget pressure. There are more open ends like that. But I can add this if you want.

> - Section/delete links to 404 confirmation-delete-news-item.htm (should be "section"!)
  MVP solution is to give a message (we have two messages: "You must first move all news items to another section before you can delete this section." and "This is the only news section and cannot be deleted. Create another section before deleting this one.")

This dialogue is still on my todo list

> Must read (are sorted on top in magazine/overview but only when unread):
> - Consumer magazine/overview: indicator + "have read" setter 
> - Editor magazine/overview indicator
> - Editor edit checkbox

> App navigation:
> - Should sort on and show must-read status, publication date and publication state instead of description

Are you quite sure about the instead?  I definitely see an added value to the description here. I could add date and state to it.

  since the combination of these 3 parameters controls the display and sequence of news items
> - Future publication date marked visually, whole item stands out as future published
> - Unpublished state indicated, whole item stands out as unpublished, when undated marked (no date) sorted on top


> Blockquote:
> - Styled in "view" but not styled in "edit" mode - maybe not in prototype but ask JC to tweak Raptor

I thought I had some basic styling for blockquotes also in raptor, but i'll have a look. It's not in the field of JC. 

> Breadcrumb is confusing if you click from "all news" into an item and then get "Back to press releases".
Better is to keep the section navbar with the current section highlighted.

I assume you refer to the back link? I don't intend to use the breadcrubss (top left next to logo) for deep content structures. Adding sub nav was a consideration. Bit clunky on mobile to get that each time on top though, this seemed more elegant, but on the other hand, don't really have anything deeply against it.

**Comment by cornae** [comment]
I'll close the ticket now if you don't mind Guido as I think part of the issue are already solved and the news app doesn't have the highest prio at this moment.

**Comment by gyst** [comment]
We will start working on this as part of the upcoming Arnhem sprint 13-17 july. Reopening. Do you prefer me to split this into separate tickets?

**Comment by cornae** [comment]
Yes, it's very hard to work with large umbrella tickets. See my complicated effort to provide inline comments on separate issues ;).

**Comment by gyst** [comment]
@cornae I assigned this ticket to myself and will split it out. Later.
